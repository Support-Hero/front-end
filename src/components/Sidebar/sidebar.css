

@media screen and (max-width:1300px)  {
   
    #sidebar {
        display: none;
    }
    #lg_main_bar{
        display: none;
    }
    #main_bar{
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
    }
}

@media screen and (min-width:1300px) {
    #sidebar_btn{
        width:150px;
    }
    #sidebar_btn:hover{
        transform: translateX(20px);
    }
    #sidebar {
        display: block;
    }
    #lg_main_bar{
        display: block;
    }
    #main_bar{
        display: none;
    }
}