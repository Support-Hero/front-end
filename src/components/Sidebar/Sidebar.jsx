import React, { useState } from 'react'
import './sidebar.css'
import { Link } from "react-router-dom";
import { lists } from '../../landpage/welcome';

const Sidebar = () => {
    const [value, setValue]=useState(-1)
    return (
        <div  id='sidebar' style={{ width: "10%",marginTop:"10%"}}>
            {lists.map((list, index) => (
                <div
                    aria-label='sidebar_link'
                    className={`mt-3 border border-${list.bgcolor} bg-${list.bgcolor} bg-opacity-75`}
                    key={index}
                    onClick={(e)=>{e.preventDefault();setValue(index)}}
                    style={{transform:value===index?"translate(10%,0)":"",}}
                >
                    <div
                        className={`d-flex justify-content-start align-items-center `}
                    >
                        <i className={`bi ${list.imsrc} text-white fs-2`}></i>
                        <Link to={`/${list.title}`}> <label className="fs-5 text-white" style={{marginLeft:'10px'}}>{list.title}</label></Link>
                    </div>

                </div>
            ))}
        </div>
    )
}
export default Sidebar