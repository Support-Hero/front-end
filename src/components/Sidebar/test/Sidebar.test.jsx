import React from 'react';
import "@testing-library/jest-dom";

import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import { afterEach, describe, expect, it, vi } from 'vitest'
import Sidebar from '../Sidebar';
import { BrowserRouter } from 'react-router-dom';

describe('Sidebar Component', () => {
//   it('renders correctly', () => {
//     const { container } = render(<BrowserRouter>
//         <Sidebar />
//         </BrowserRouter>)
//     expect(container).toMatchSnapshot();
//   });

    it("should render side link bar",()=>{
        render(<BrowserRouter>
        <Sidebar />
        </BrowserRouter>)
        expect(screen.getAllByLabelText("sidebar_link")).toHaveLength(4)
    })
  it('changes value when a list item is clicked', () => {
    const { getByText, getByTestId } = render(
    <BrowserRouter>
        <Sidebar />
        </BrowserRouter>)

    const listItem = getByText('workers'); // Replace with an actual list title
    fireEvent.click(listItem);

    const sidebar = screen.getAllByLabelText("sidebar_link")
    // expect(sidebar).toHaveStyle('transform: translate(10%, 0)');
  });
});