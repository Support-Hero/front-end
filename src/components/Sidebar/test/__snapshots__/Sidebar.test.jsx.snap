// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sidebar Component > renders correctly 1`] = `
<div>
  <div
    id="sidebar"
    style="width: 10%; margin-top: 10%;"
  >
    <div
      aria-label="sidebar_link"
      class="mt-3 border border-primary  bg-primary  bg-opacity-75"
    >
      <div
        class="d-flex justify-content-start align-items-center "
      >
        <i
          class="bi bi-people text-white fs-2"
        />
        <a
          href="/workers"
        >
           
          <label
            class="fs-5 text-white"
            style="margin-left: 10px;"
          >
            workers
          </label>
        </a>
      </div>
    </div>
    <div
      aria-label="sidebar_link"
      class="mt-3 border border-success  bg-success  bg-opacity-75"
    >
      <div
        class="d-flex justify-content-start align-items-center "
      >
        <i
          class="bi bi-person-vcard text-white fs-2"
        />
        <a
          href="/clients"
        >
           
          <label
            class="fs-5 text-white"
            style="margin-left: 10px;"
          >
            clients
          </label>
        </a>
      </div>
    </div>
    <div
      aria-label="sidebar_link"
      class="mt-3 border border-secondary  bg-secondary  bg-opacity-75"
    >
      <div
        class="d-flex justify-content-start align-items-center "
      >
        <i
          class="bi bi-file-earmark-medical text-white fs-2"
        />
        <a
          href="/rosters"
        >
           
          <label
            class="fs-5 text-white"
            style="margin-left: 10px;"
          >
            rosters
          </label>
        </a>
      </div>
    </div>
    <div
      aria-label="sidebar_link"
      class="mt-3 border border-danger bg-danger bg-opacity-75"
    >
      <div
        class="d-flex justify-content-start align-items-center "
      >
        <i
          class="bi bi-microsoft-teams text-white fs-2"
        />
        <a
          href="/notes"
        >
           
          <label
            class="fs-5 text-white"
            style="margin-left: 10px;"
          >
            notes
          </label>
        </a>
      </div>
    </div>
  </div>
</div>
`;
